### PR TITLE
Implementation of MSC3882 login token request

### PR DESCRIPTION
**Pull request by hughns**
As the added functionality is just passing through an HTTP request I don't think unit tests are needed for this.

## Checklist

* [x] Tests written for new code (and old code if feasible)
* [x] Linter and other CI checks pass
* [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * Implementation of MSC3882 login token request ([\#2687](https://github.com/matrix-org/matrix-js-sdk/pull/2687)). Contributed by @hughns.

### CODE REVIEW

File: src/@types/uia.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { IAuthData } from "..";
```
Comment (SimonBrandner) [inline src/@types/uia.ts:1]: License please

File: src/client.ts [hunk 2]
```diff
@@ -7277,6 +7278,20 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
...
+     * [MSC3882](https://github.com/matrix-org/matrix-spec-proposals/pull/3882).
+     * The server may require User-Interactive auth.
+     * Note that this is UNSTABLE and subject to breaking changes without notice.
+     * @param {IAuthData} auth Optional. Auth data to supply for User-Interactive auth.
+     * @return {Promise<UIAResponse<ILoginTokenPostResponse>>} Resolves: On success, the token response
+     * or UIA auth data.
+     */
+    public requestLoginToken(auth?: IAuthData): Promise<UIAResponse<ILoginTokenPostResponse>> {
+        const body: UIARequest<{}> = { auth };
+        return this.http.authedRequest(undefined, Method.Post, "/login/token", undefined, body);
```
Comment (SimonBrandner) [inline src/client.ts:7292]: You say this is unstable but the endpoint being used looks stable?

**Review by SimonBrandner — COMMENTED** [review]
LGTM otherwise

File: src/@types/auth.ts [hunk 3]
```diff
@@ -90,3 +90,19 @@ export enum SSOAction {
     /** The user intends to register for a new account */
     REGISTER = "register",
 }
+
+/**
+ * The result of a successful [MSC3882](https://github.com/matrix-org/matrix-spec-proposals/pull/3882)
+ * `m.login.token` issuance request.
+ * Note that this is UNSTABLE and subject to breaking changes without notice.
+ */
+export interface ILoginTokenPostResponse {
```
Comment (weeman1337) [inline src/@types/auth.ts:99]: We should not prefix interfaces with `I` according to our new [code style](https://github.com/vector-im/element-web/blob/develop/code_style.md)

**Review by SimonBrandner — APPROVED** [review]

**Review by weeman1337 — APPROVED** [review]
LGTM now :+1: 

Could still use some tests for code coverage :wink:

**Comment by hughns** [comment]
@weeman1337 tests now added too. 🤞

**Pull request merged**